### ISSUE TITLE
Cannot find -lopencv_core401 

### ISSUE THREAD

**Issue by paruliansaragi** [hybridgroup/gocv#472]
Receiving a list of cannot find -l<> error messages.

```
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_core401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_face401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_videoio401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_imgproc401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_highgui401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_imgcodecs401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_objdetect401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_features2d401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_video401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_dnn401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_xfeatures2d401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_plot401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_tracking401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_img_hash401
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopencv_calib3d401
```

## Description
I believe I'm getting a lot of linker errors but I'm not sure as to why since I followed the installation. I started a new project and I need to import gocv. 
I installed Mingw and MSYS differently [here](https://github.com/Kagami/go-face#windows).

## Steps to Reproduce

1. Follow the Install which works as expected.
2. The cmd/version/main.go runs fine
3. Create a new workspace/directory
4. Import gocv.io/x/gocv
5. Run main.go 

## Your Environment

* Operating System and version:
- Windows 10
* OpenCV version used:
- 4.1.0
* How did you install OpenCV?
- I followed the installation instructions for windows via MSYS and Mingw64 although I did not use the installation here for Mingw and MSYS 

* Did you run the `env.sh` or `env.cmd` script before trying to `go run` or `go build`?
- did not run the `env.cmd` script and I did not set up the custom environment tags.

**Comment by paruliansaragi** [comment]
Seems to be an issue with using gocv in a location other than the `C:\msys64\mingw64\src\gocv.io\x\gocv\cmd` directory. If I place code into the aforementioned directory it seems to run fine but not outside.

**Comment by paruliansaragi** [comment]
I modified the `win_build_opencv.cmd` file to point to my `C:\msys64\mingw64\bin` directory.

**Comment by WendelHime** [comment]
Hi @paruliansaragi , you're using gocv dev or master branch?

**Comment by paruliansaragi** [comment]
I assume master, I pulled go via `go get -u -d gocv.io/x/gocv`

**Comment by WendelHime** [comment]
strange, it shouldn't search for opencv libs from the version 4.0.1, you've pulled 2 days ago or before 21/05/2019 (deadprogram updated `win_build_opencv.cmd` with the version 4.1 on this date)?

Anyway, the modification on `win_build_opencv.cmd` fixed the problem?

**Comment by paruliansaragi** [comment]
No it didn't work. It did pull the version 4.0.1 and compiled that. The library works only if your inside `C:\msys64\mingw64\src\gocv.io\x\gocv\cmd`. 

My modification to the `win_build_opencv.cmd` only managed to enable the `cmake` and `mingw32_make` instructions to work. 

I think it may be something to do with MinGW since I did not use `mingw-w64\x86_64-7.3.0-posix-seh-rt_v5-rev2` but I used MSYS to install MinGW via pacman. 

Do I need to edit the CMake lists.txt?

**Comment by paruliansaragi** [comment]
I'm trying to get it to work [with](https://github.com/Kagami/go-face) so I used MSYS to install dlib. 

Would you advise re-doing with the `mingw-w64\x86_64-7.3.0-posix-seh-rt_v5-rev2` and re-building with the changes from 4.0.1 to 4.1?

**Comment by WendelHime** [comment]
I'd advise to rebuild OpenCV things from 4.0.1 to 4.1, I guess it will solve

**Comment by entrehuihui** [comment]
I solved this problem after installing 4.1.

**Comment by deadprogram** [comment]
The most recent release of GoCV requires OpenCV 4.1.0

You will need to upgrade your OpenCV install in order to use it.

Hope that helps! Now closing.
